firewall.go

```diff
@@ -0,0 +1,246 @@
+	return api.updateAccessRule("/zones/"+zoneID, accessRuleID, accessRule)
+}
+
+// DeleteZoneAccessRule deletes a single access rule for the given zone and
+// access rule identifiers.
+//
+// API reference: https://api.cloudflare.com/#firewall-access-rule-for-a-zone-delete-access-rule
+func (api *API) DeleteZoneAccessRule(zoneID, accessRuleID string) error {
+	return api.deleteAccessRule("/zones/"+zoneID, accessRuleID)
+}
+
+// ListOrganizationAccessRules returns a slice of access rules for the given
+// organization identifier.
+//
+// This takes an AccessRule to allow filtering of the results returned.
+//
+// API reference: https://api.cloudflare.com/#organization-level-firewall-access-rule-list-access-rules
+func (api *API) ListOrganizationAccessRules(organizationID string, accessRule AccessRule) ([]AccessRule, error) {
+	return api.listAccessRules("/organizations/"+organizationID, accessRule)
+}
+
+// CreateOrganizationAccessRule creates a firewall access rule for the given
+// organization identifier.
+//
+// API reference: https://api.cloudflare.com/#organization-level-firewall-access-rule-create-access-rule
+func (api *API) CreateOrganizationAccessRule(organizationID string, accessRule AccessRule) (*AccessRuleResponse, error) {
+	return api.createAccessRule("/organizations/"+organizationID, accessRule)
+}
+
+// UpdateOrganizationAccessRule updates a single access rule for the given
+// organization & access rule identifiers.
+//
+// API reference: https://api.cloudflare.com/#organization-level-firewall-access-rule-update-access-rule
+func (api *API) UpdateOrganizationAccessRule(organizationID, accessRuleID string, accessRule AccessRule) error {
+	return api.updateAccessRule("/organizations/"+organizationID, accessRuleID, accessRule)
+}
+
+// DeleteOrganizationAccessRule deletes a single access rule for the given
+// organization and access rule identifiers.
+//
+// API reference: https://api.cloudflare.com/#organization-level-firewall-access-rule-delete-access-rule
+func (api *API) DeleteOrganizationAccessRule(organizationID, accessRuleID string) error {
+	return api.deleteAccessRule("/organizations/"+organizationID, accessRuleID)
+}
+
+func (api *API) listAccessRules(prefix string, accessRule AccessRule) ([]AccessRule, error) {
+	// Construct a query string
+	v := url.Values{}
+	// Request as many rules as possible per page - API max is 100
+	v.Set("per_page", "100")
+	if accessRule.Notes != "" {
+		v.Set("notes", accessRule.Notes)
+	}
+	if accessRule.Mode != "" {
+		v.Set("mode", accessRule.Mode)
+	}
+	if accessRule.Scope.Type != "" {
+		v.Set("scope_type", accessRule.Scope.Type)
+	}
+	if accessRule.Configuration.Value != "" {
+		v.Set("configuration_value", accessRule.Configuration.Value)
+	}
+	if accessRule.Configuration.Target != "" {
+		v.Set("configuration_target", accessRule.Configuration.Target)
+	}
+
+	var query string
+	var rules []AccessRule
+	page := 1
+
+	// Loop over makeRequest until what we've fetched all rules
```

Comment:
It may make more sense to take a `page int` arg to `ListAccessRules` instead, and allow the library consumer to call `client.ListZoneAccessRules` on their own if they need to paginate. This is what I did for ListZoneLockdowns: https://godoc.org/github.com/cloudflare/cloudflare-go#API.ListZoneLockdowns

This is probably esp. useful when you have ~10,000 rules and are fetching 100 at a time; as a consumer you can also create a `sync.WaitGroup` or worker pool and fire off the fetches concurrently to speed things up.

Reply:
Easy enough to do. I will make that adjustment. I also just noticed that I didn't update flarectl, so I'll do that, too.

Reply:
🙌🏼